public/app/plugins/datasource/tempo/streaming.ts

```diff
@@ -100,6 +108,120 @@ export function doTempoChannelStream(
+  options: DataQueryRequest<TempoQuery>
+): Observable<DataQueryResponse> {
+  const range = options.range;
+  const key = getLiveStreamKey();
+
+  let state: LoadingState = LoadingState.NotStarted;
+  const step = stepToNanos(query.step);
+
+  return getGrafanaLiveSrv()
+    .getStream<MutableDataFrame>({
+      scope: LiveChannelScope.DataSource,
+      namespace: ds.uid,
+      path: `metrics/${key}`,
+      data: {
+        ...query,
+        step,
+        timeRange: {
+          from: range.from.toISOString(),
+          to: range.to.toISOString(),
+        },
+      },
+    })
+    .pipe(
+      takeWhile((evt) => {
+        if ('message' in evt && evt?.message) {
+          const frameState: SearchStreamingState = evt.message.data.values[2][0];
+          if (frameState === SearchStreamingState.Done || frameState === SearchStreamingState.Error) {
+            return false;
+          }
+        }
+        return true;
+      }, true),
+      map((evt) => {
+        let newResult: DataQueryResponse = { data: [], state: LoadingState.NotStarted };
+        if ('message' in evt && evt?.message) {
+          const messageFrame = dataFrameFromJSON(evt.message);
+          const fieldCache = new FieldCache(messageFrame);
+
+          const data = fieldCache.getFieldByName('result')?.values[0];
+          const frameState = fieldCache.getFieldByName('state')?.values[0];
+          const error = fieldCache.getFieldByName('error')?.values[0];
+
+          switch (frameState) {
+            case SearchStreamingState.Done:
+              state = LoadingState.Done;
+              break;
+            case SearchStreamingState.Streaming:
+              state = LoadingState.Streaming;
+              break;
+            case SearchStreamingState.Error:
+              throw new Error(error);
+          }
+
+          newResult = {
+            data: data?.map(dataFrameFromJSON) ?? [],
+            state,
+          };
+        }
+
+        return newResult;
+      }),
+      // Merge results on acc
+      scan((acc, curr) => {
+        if (!curr) {
+          return acc;
+        }
+        if (!acc) {
+          return cloneQueryResponse(curr);
+        }
+        return mergeFrames(acc, curr);
+      })
+    );
+}
+
+function mergeFrames(acc: DataQueryResponse, newResult: DataQueryResponse): DataQueryResponse {
+  const result = combineResponses(cloneQueryResponse(acc), newResult);
+
+  // Remove duplicate time field values for all frames
+  result.data = result.data.map((frame: DataFrame) => {
+    let newFrame = frame;
+    const timeFieldIndex = frame.fields.findIndex((f) => f.type === FieldType.time);
+    if (timeFieldIndex >= 0) {
+      removeDuplicateTimeFieldValues(frame, timeFieldIndex);
+      newFrame = sortDataFrame(frame, timeFieldIndex);
+    }
+    return newFrame;
+  });
+
+  result.state = newResult.state;
+  return result;
+}
+
+function removeDuplicateTimeFieldValues(accFrame: DataFrame, timeFieldIndex: number) {
```

Comment:
Could you add a comment as to why this is necessary so we don't forget in the future.